next-cmake/device-library/CMakeLists.txt

```diff
@@ -42,18 +42,22 @@ set(TENSILE_CREATE_LIBRARY_COMMAND
     HIP
 )
 
-add_custom_target(tensilelite-device-libraries ALL
+add_custom_command(
+    OUTPUT "${output_stamp}"
     COMMENT "Building device libraries to ${output_dir} ..."
     COMMAND ${TENSILE_CREATE_LIBRARY_COMMAND}
     COMMAND ${CMAKE_COMMAND} -E touch "${output_stamp}"
-    BYPRODUCTS "${output_stamp}"
-    DEPENDS ${HIPBLASLT_PYTHON_DEPS}
+    DEPENDS  ${HIPBLASLT_PYTHON_DEPS}
     # Because the command can contain special characters
     VERBATIM
     # Because this can be very long running and difficult to debug deadlocks
     # without streaming.
     USES_TERMINAL
 )
 
+add_custom_target(tensilelite-device-libraries ALL
```

Comment:
Thanks for fixing this - indeed in the prior work, I got too aggressive in collapsing (I was working on this due to a couple of other bugs and then was like, this can be simplified more, but was one notch too far). The way I had it was always triggering the commands since the custom target is always invalid.